src/sql/platform/connection/common/connectionStatusManager.ts

```diff
@@ -49,6 +49,19 @@ export class ConnectionStatusManager {
 		return !!this.findConnection(id);
 	}
 
+	public changeConnectionUri(newId: string, oldId: string) {
+		let info = this.findConnection(oldId);
+		if (!info) {
+			this._logService.error(`No connection found associated with old URI : '${oldId}'`);
```

Comment:
This should probably throw as well (or at the very least it needs to return to avoid null ref on line 57)